### PR TITLE
Postfix: security update to 3.2.2, add PostgreSQL and PCRE support and refresh patches

### PR DESCRIPTION
**Pull request by lucize**
switch to latest upstream, the update contains security fixes !

add option for PostgreSQl and PCRE
refresh patches
cosmetic fix

maintainer: @Shulyaka 

Compile/Run tested on x86  Lede r4588-cd10577

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

### CODE REVIEW

**Comment by champtar** [comment]
@lucize @Shulyaka I think you should have a PKG_CONFIG_DEPENDS in the Makefile
@lucize you are enabling PCRE by default, what is the size of postfix with/without it, and what is the size of the new dependency ?

**Comment by lucize** [comment]
if I remember @400KB with PCRE and PGSQL, do we have space problems on systems that are running postfix ?
regards

**Comment by champtar** [comment]
@lucize no idea who is using postfix on LEDE ;), but also I don't know who wants PCRE by default, having the size can help @Shulyaka take a decision

**Comment by lucize** [comment]
LDAP is also by default, LEDE is ideal for a smarthost and spam/antivirus gateway
the size is the same, I think because is already static linked to libpcre
-rw-r--r--  1 build build 4170404 Jul 17 01:32 pcre-postfix_3.2.2-1_x86_64.ipk
-rw-r--r--  1 build build 4170404 Jul 17 01:34 postfix_3.2.2-1_x86_64.ipk

**Comment by champtar** [comment]
@lucize
exactly the same size is really suspect, I'm pretty sure if you do ```make package/postfix/clean``` you will find different size, and then google PKG_CONFIG_DEPENDS and add it ;)

**Comment by lucize** [comment]
yes, forgot about clean
-rw-r--r--  1 build build 4170404 Jul 17 01:32 pcre-postfix_3.2.2-1_x86_64.ipk
-rw-r--r--  1 build build 4055063 Jul 17 01:49 postfix_3.2.2-1_x86_64.ipk
115,341 diff
PKG_CONFIG_DEPENDS later

**Comment by lucize** [comment]
added PKG_CONFIG_DEPENDS, hopefully I understood what is there for and disabled default PCRE

**Comment by champtar** [comment]
Hi @lucize 
a good exemple https://github.com/lede-project/source/blob/master/package/libs/openssl/Makefile#L31
PKG_CONFIG_DEPENDS means rerun the configure step if one of the following config options changes, so you need to put the package config options (CONFIG_POSTFIX_TLS ...), not the dependencies

**Comment by lucize** [comment]
ok, updated and squashed

**Comment by champtar** [comment]
Thanks @lucize, waiting for @Shulyaka now

File: mail/postfix/Makefile [hunk 1]
```diff
@@ -64,16 +74,26 @@ define Package/postfix/config
...
 			bool "SMTPUTF8 support"
 			default n
 			help
-				Enable Postfix support for Email Address Internationalization
-				(EAI) as defined in RFC 6531 (SMTPUTF8 extension), RFC 6532
-				(Internationalized email headers) and RFC 6533
-				(Internationalized delivery status notifications).
-				Since version 3.0, Postfix fully supports UTF-8 email
-				addresses and UTF-8 message header values.
+			  Enable Postfix support for Email Address Internationalization
```
Comment (pprindeville) [inline mail/postfix/Makefile]: Please remove the leading space (singular) here.

File: mail/postfix/Makefile [hunk 2]
```diff
@@ -25,7 +35,7 @@ define Package/postfix
   CATEGORY:=Mail
   TITLE:=Postfix Mail Transmit Agent
   URL:=http://www.postfix.org/
-  DEPENDS:=+POSTFIX_TLS:libopenssl +POSTFIX_SASL:libsasl2 +POSTFIX_LDAP:libopenldap +POSTFIX_DB:libdb47 +POSTFIX_SQLITE:libsqlite3 +POSTFIX_EAI:icu +libpcre
+  DEPENDS:=+POSTFIX_TLS:libopenssl +POSTFIX_SASL:libsasl2 +POSTFIX_LDAP:libopenldap +POSTFIX_DB:libdb47 +POSTFIX_SQLITE:libsqlite3 +POSTFIX_PGSQL:libpq +POSTFIX_EAI:icu +libpcre
```
Comment (pprindeville) [inline mail/postfix/Makefile]: Should it be `+POSTFIX_PCRE:libpcre` and not `+libpcre`?

File: mail/postfix/patches/300-bdb_hash_segfault.patch [hunk 3]
```diff
@@ -1,10 +1,9 @@
-diff -Naur postfix-2.11.3/src/util/dict_db.c postfix-2.11.3_patched/src/util/dict_db.c
---- postfix-2.11.3/src/util/dict_db.c	2012-01-25 04:41:08.000000000 +0400
-+++ postfix-2.11.3_patched/src/util/dict_db.c	2014-11-01 12:36:44.287641712 +0300
-@@ -687,8 +687,8 @@
+--- a/src/util/dict_db.c	2017-07-16 16:56:00.819659962 +0300
```
Comment (pprindeville) [inline mail/postfix/patches/300-bdb_hash_segfault.patch]: What happened here?  More than just the line numbers getting updated by a quilt refresh...

Comment (lucize) [inline mail/postfix/patches/300-bdb_hash_segfault.patch]: I only changed the line refferences

File: mail/postfix/Makefile (hunk 2 above)
Comment (lucize) [inline mail/postfix/Makefile]: the base will not build without libpcre, a better approach would be to dynamic linking but couldn't make it work (not enough knowledge)

File: mail/postfix/Makefile (hunk 1 above)
Comment (lucize) [inline mail/postfix/Makefile]: yeah, a mistake

Comment (pprindeville) [inline mail/postfix/Makefile]: You have tab-tab-space.  Please make it tab-tab.

File: mail/postfix/Makefile (hunk 2 above)
Comment (pprindeville) [inline mail/postfix/Makefile]: Then I don't understand what the point of having `POSTFIX_PCRE` below is, if it (`libpcre`) needs to be included unconditionally...

**Comment by lucize** [comment]
me neither, but without specifying  DHAS_PCRE it will not build with PCRE support, not specifying PCRE the binary is smaller, as the above size tests

**Comment by lucize** [comment]
it should be possible to -DNO_PCRE I'll make a pull later

File: mail/postfix/Makefile (hunk 1 above)
Comment (lucize) [inline mail/postfix/Makefile]: yes I know, is tab-tab-space-space like other help information on the file, maybe should I make tab-tab for all the help statements ?

Comment (pprindeville) [inline mail/postfix/Makefile]: No, leave it as is. I withdraw the comment.

File: mail/postfix/Makefile [hunk 4]
```diff
@@ -116,6 +136,16 @@ ifdef CONFIG_POSTFIX_SQLITE
   AUXLIBS+=-L$(STAGING_DIR)/usr/lib -lsqlite3 -lpthread
 endif
 
+ifdef CONFIG_POSTFIX_PGSQL
+  CCARGS+=-DHAS_PGSQL -I$(STAGING_DIR)/usr/include/
+  AUXLIBS+=-L$(STAGING_DIR)/usr/lib -lpq
+endif
+
+ifdef CONFIG_POSTFIX_PCRE
+  CCARGS+=-DHAS_PCRE -I$(STAGING_DIR)/usr/include/
```
Comment (pprindeville) [inline mail/postfix/Makefile]: Do we need `-DHAS_PCRE` in the affirmative case?  The build directions:

http://www.postfix.org/INSTALL.html#build_other

seems to imply that we only need to call it out in the negative case: `-DNO_PCRE`.

Comment (lucize) [inline mail/postfix/Makefile]: for sure adding -DHAS_PCRE adds ~100kb to the compressed ipk file, didn't had time to test for functionality on PCRE, but I was asked to explicit add PCRE possibility, so it seems that before it didn't worked

File: mail/postfix/Makefile [hunk 5]
```diff
@@ -116,6 +136,16 @@ ifdef CONFIG_POSTFIX_SQLITE
...
+  AUXLIBS+=-L$(STAGING_DIR)/usr/lib -lpcre
+endif
```
Comment (pprindeville) [inline mail/postfix/Makefile]:
```
else
  CCARGS+=-DNO_PCRE
```

File: mail/postfix/patches/300-bdb_hash_segfault.patch (hunk 3 above)
Comment (pprindeville) [inline mail/postfix/patches/300-bdb_hash_segfault.patch]: Okay

File: mail/postfix/Makefile (hunk 4 above)
Comment (lucize) [inline mail/postfix/Makefile]: updated
-rw-r--r--  1 build build 4170404 Jul 18 15:45 pcre-postfix_3.2.2-1_x86_64.ipk
-rw-r--r--  1 build build 4055197 Jul 18 15:40 postfix_3.2.2-1_x86_64.ipk

**Comment by lucize** [comment]
@champtar @hnyman is good to be merged now

**Comment by pprindeville** [comment]
Shall I merge or let @hnyman do it?

**Comment by lucize** [comment]
anybody who has the rights..

regards !

**Comment by pprindeville** [comment]
Merging.  @lucize can you get in contact with @val-kulkov who is working on packaging Sendmail's `libmilter` which Postfix can also link to?  Thanks.

**Comment by lucize** [comment]
yes, I saw the libmilter and hoped to get merged so it can be used with clamav and no more clam-smtp, I'll try to talk with him to finish the commit

**Comment by pprindeville** [comment]
Awesome.  Good luck to both of you.

**Comment by val-kulkov** [comment]
@lucize , @pprindeville : as I got nudged into building all of sendmail, not just the libmilter package, I am going to create a new PR and close PR #4369 as soon as the new PR for sendmail + libmilter + vacation + makemap etc is ready. It is taking some time. If it were just the libmilter package, it would have been done already.

**Comment by pprindeville** [comment]
Well, the nudge was towards getting a top-level make to work, since that's what's typically done (and it tends to be more reliable).

If you want to only generate a `sendmail-milter` package and not have a parent `sendmail` package and no sibling packages for now, that's fine.  It's your call if you circle back later to round out the other package/subpackages later or let someone else do that.

**Comment by champtar** [comment]
@pprindeville I know @Shulyaka was not reactive, but if you don't wait for maintainer ack/nack, what is the point of having maintainers ...

**Comment by lucize** [comment]
@Shulyaka agreed with changes, @pprindeville just wanted an improvement

**Comment by champtar** [comment]
sorry @pprindeville, missed the fact that @Shulyaka indeed approved ...

**Comment by pprindeville** [comment]
> sorry @pprindeville, missed the fact that @Shulyaka indeed approved ...

Pas de probleme!

**Comment by Shulyaka** [comment]
Yes I did, sorry for not being very reactive:)

вт, 18 июля 2017, 23:36 champtar <notifications@github.com>:

**Comment by champtar** [comment]
@Shulyaka i though you didn't responded at all ...
1 day is reactive enough

**Pull request merged**